modules/evm/src/lib.rs

```diff
@@ -956,9 +970,59 @@ impl<T: Config> Pallet<T> {
 			}
 		});
 
-		frame_system::Pallet::<T>::inc_providers(&T::AddressMapping::get_account_id(address));
+		frame_system::Pallet::<T>::inc_providers(&T::AddressMapping::get_account_id(&address));
+	}
 
-		Ok(())
+	/// Get the account basic in EVM format.
+	pub fn account_basic(address: &EvmAddress) -> Account {
+		let account_id = T::AddressMapping::get_account_id(address);
+
+		let nonce = Self::accounts(address).map_or(Default::default(), |account_info| account_info.nonce);
+		let balance = T::Currency::free_balance(&account_id);
+
+		Account {
+			nonce: U256::from(UniqueSaturatedInto::<u128>::unique_saturated_into(nonce)),
+			balance: U256::from(UniqueSaturatedInto::<u128>::unique_saturated_into(balance)),
+		}
+	}
+
+	/// Get the author using the FindAuthor trait.
+	pub fn find_author() -> H160 {
+		let digest = <frame_system::Pallet<T>>::digest();
+		let pre_runtime_digests = digest.logs.iter().filter_map(|d| d.as_pre_runtime());
+
+		T::FindAuthor::find_author(pre_runtime_digests).unwrap_or_default()
+	}
+
+	/// Get code hash at given address.
+	pub fn code_hash_at_address(address: &EvmAddress) -> H256 {
+		if let Some(AccountInfo {
+			contract_info: Some(contract_info),
+			..
+		}) = Self::accounts(address)
+		{
+			contract_info.code_hash
+		} else {
+			code_hash(&[])
```

Comment:
better to precompute this and hardcode it, and some unit test to test the hardcoded value